packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -212,6 +212,14 @@ export class FluidDataStoreRuntime extends EventEmitter implements IFluidDataSto
                         (content, localOpMetadata) => this.submitChannelOp(path, content, localOpMetadata),
                         (address: string) => this.setChannelDirty(address),
                         tree.trees[path]);
+                    // If the data store is loaded after the container is attached, then attach the channel right away.
+                    // Otherwise add it to local channel context queue, so that it can be attached later with the data
+                    // store.
+                    if (dataStoreContext.attachState !== AttachState.Detached) {
```

Comment:
It took me a while to understand why it's right. Not sure if we can improve comments, but please consider that.
The key here is that we serialize & de-serialize only bound data stores (which we might need to change, not sure - see Issue #3647).
So context in such case will get attached when container is attached.
Might be worth adding a bit more comments here about it (will be useful as we evaluate above mentioned issue)

Should "attach()" be renamed to "markAttached()" or something like that?
attach() IMHO has a lot of connation to attachGraph, which is action, while here there is no action, just mark up of object.

Reply:
I'm confused. i thought only the container's attached to storage, and everything else was bound to it's parent

---
In reply to: [489152975](https://github.com/microsoft/FluidFramework/pull/3645#discussion_r489152975) [](ancestors = 489152975)

Reply:
@anthony-murphy Yes you are right. When the containers attach to storage, everything bound to it gets attached to storage. 
In this case also, it is actually present on the storage and therefore already attached but due to delay loading of the data store, channel was not mark attached as vlad said in his comment. So here we are just marking it up.

Reply:
So I think I will take vlad suggestion and rename the api.